pkg/kube/controllers/quarksjob/service_account.go

```diff
@@ -0,0 +1,95 @@
+package quarksjob
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/rbac/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	crc "sigs.k8s.io/controller-runtime/pkg/client"
+
+	"code.cloudfoundry.org/quarks-utils/pkg/names"
+	"code.cloudfoundry.org/quarks-utils/pkg/pointers"
+)
+
+const (
+	serviceAccountName            = "persist-output-service-account"
+	serviceAccountSecretMountPath = "/var/run/secrets/kubernetes.io/serviceaccount"
+)
+
+func (j jobCreatorImpl) createdServiceAccount(ctx context.Context, namespace string) (*corev1.Volume, *corev1.VolumeMount, error) {
```

Comment:
Why should we return pointers?

Reply:
I'm a fan of returning an empty struct, but I thought everyone else prefers returning nil with errors?

You can still call methods on nil, so can I use empty struct?